concent_api/gatekeeper/views.py

```diff
@@ -124,73 +161,193 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key
+        )
     transfer_token_paths_to_files = [file["path"] for file in loaded_golem_message.files]
     if len(transfer_token_paths_to_files) != len(set(transfer_token_paths_to_files)):
-        return gatekeeper_access_denied_response('File paths in the token must be unique', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            'File paths in the token must be unique',
+            ErrorCode.MESSAGE_FILES_PATH_NOT_UNIQUE.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
 
     if not all(isinstance(file["checksum"], str) for file in loaded_golem_message.files):
-        return gatekeeper_access_denied_response("'checksum' must be a string.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "'checksum' must be a string.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_TYPE.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     if any((len(file["checksum"]) == 0 or file["checksum"].isspace()) for file in loaded_golem_message.files):
-        return gatekeeper_access_denied_response("'checksum' cannot be blank or contain only whitespace.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "'checksum' cannot be blank or contain only whitespace.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_EMPTY.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     if not all(":" in file["checksum"] for file in loaded_golem_message.files):
-        return gatekeeper_access_denied_response("'checksum' must consist of two parts separated with a semicolon.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "'checksum' must consist of two parts separated with a semicolon.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_FORMAT.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     file_checksums = [tuple(file["checksum"].split(":")) for file in loaded_golem_message.files]
     if not set(checksum_type for checksum_type, checksum in file_checksums).issubset(set(HashingAlgorithm._value2member_map_)):  # type: ignore
-        return gatekeeper_access_denied_response("Unrecognized checksum type.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "Unrecognized checksum type.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_UNRECOGNIZED_TYPE.value,
```

Comment:
`Unrecognized checksum type` -> `One of the checksums is from an unsupported hashing algorithm.`
`MESSAGE_FILES_CHECKSUM_UNRECOGNIZED_TYPE` -> `MESSAGE_FILES_CHECKSUM_INVALID_ALGORITHM`

Reply:
:+1: